examples/usematches-loader-data/app/useLoaderStore.ts

```diff
@@ -0,0 +1,20 @@
+import { useMemo } from "react";
+import { useMatches } from "remix";
+
+/**
+ * This base hook is used in other hooks to quickly search for specific data
+ * across all loader data using useMatches.
+ * @param {string} key
+ * @returns {unknown} The value of the key based on your currently loaded routes.
+ */
+export default function useLoaderStore(key: string): unknown {
```

Comment:
Interesting approach.

I don't think I'm a fan, because you could have an unexpected behavior if you had two parent loaders that use the same key (for example, you have a `/users/$userId` where that loader returns a `user` for that route).

I'd rather have something like:

```suggestion
export default function useParentLoaderData(id: string): unknown {
```

Where the `id` is the ID of the route and it gives you *all* the loader data for that route.

Reply:
Thank you for the review and the detailed change request! I will adapt the code and push a commit. 

Just one follow-up question @kentcdodds : The naming `useParentLoaderData` could be confusing as you can theoretically also access data of your child routes, right? Imagine a global nav component that displays a more detailed menu once the project details page returns the `project` object from its loader. So it could also use the `useParentLoaderData` hook to access optional data from its child routes. That might be bad practice though. Your call!

Reply:
Maybe `useMatchesLoaderData`?